lighthouse-core/test/audits/preload-fonts-test.js

```diff
@@ -0,0 +1,214 @@
+          isLinkPreload: true,
+        },
+      ];
+
+      const result = await PreloadFontsAudit.audit(getArtifacts(), context);
+      expect(result.details.items).toEqual([]);
+      expect(result.score).toEqual(1);
+    });
+  });
+
+  describe('font-display is not optional', () => {
+    it('passes if the font is not preloaded', async () => {
+      stylesheet.content = `
+        @font-face {
+          src: url('/assets/font-a.woff');
+        }
+      `;
+
+      networkRecords = [
+        {
+          url: 'https://example.com/assets/font-a.woff',
+          endTime: 3, startTime: 1,
+          resourceType: 'Font',
+          isLinkPreload: false,
+        },
+      ];
+
+      const result = await PreloadFontsAudit.audit(getArtifacts(), context);
+      expect(result.score).toEqual(1);
+      expect(result.details.items).toEqual([]);
+      expect(result.notApplicable).toEqual(true);
+    });
+
+    it('passes if the font is preloaded', async () => {
+      stylesheet.content = `
+        @font-face {
+          src: url('/assets/font-a.woff');
+        }
+      `;
+
+      networkRecords = [
+        {
+          url: 'https://example.com/assets/font-a.woff',
+          endTime: 3, startTime: 1,
+          resourceType: 'Font',
+          isLinkPreload: true,
+        },
+      ];
+
+      const result = await PreloadFontsAudit.audit(getArtifacts(), context);
+      expect(result.score).toEqual(1);
+      expect(result.details.items).toEqual([]);
+      expect(result.notApplicable).toEqual(true);
+    });
+  });
+
+  it('is not applicable on fonts where font-display is not optional', async () => {
+    stylesheet.content = `
+      @font-face {
+        font-display: swap;
+        src: url('/assets/font-a.woff');
+      }
+
+      @font-face {
+        font-display: block;
+        src: url('https://example.com/foo/bar/document-font.woff');
+      }
+
+      @font-face {
+        font-display: fallback;
+        src: url('/assets/font-b.woff');
+      }
+
+      @font-face {
+        src: url('/assets/font-c.woff');
+      }
+    `;
+
+    networkRecords = [];
+
+    const result = await PreloadFontsAudit.audit(getArtifacts(), context);
+    expect(result.score).toEqual(1);
+    expect(result.details.items).toEqual([]);
+    expect(result.notApplicable).toEqual(true);
+  });
+
+  it('handles multiple fonts', async () => {
+    stylesheet.content = `
+      @font-face {
+        font-display: optional;
+        src: url('/assets/font-a.woff');
+      }
+
+      @font-face {
+        font-display: optional;
+        src: url('https://example.com/foo/bar/document-font.woff');
+      }
+
+      @font-face {
+        font-display: fallback;
+        src: url('/assets/font-b.woff');
+      }
+
+      @font-face {
+        font-display: optional;
+        src: url('/assets/font-c.woff');
+      }
+    `;
+
+    networkRecords = [
+      {
+        url: 'https://example.com/assets/font-a.woff',
+        endTime: 3, startTime: 1,
```

Comment:
do all these tests still work if you delete the time properties?

Reply:
good point, I didn't check if necessary because in mocked `networkRecords` elsewhere, `endTime/startTime` were included

I'll double check and remove them if they're unnecessary!